### PR TITLE
Removes Bluespace Tile Speed, Replaces It With Anti-Slowdown

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Removes the insanely fast speedboost that Bluespace Tiles give you.

They now remove various slowdowns that the Crew deals with - giving you some decent QoL while on these tiles.

**Currently Affected**
- [x] Pulling
- [x] Suit/Gear Slowdown

**WIP Affected**

~~JaniCart/Ambulance Slowdown~~ (never mind fuck vehicle code)

## Why It's Good For The Game
The Bluespace Tiles are a massive headache. They stack with things like meth, changeling speedlegs, and xenomorphs.

I have quite literally charged Xenobiologists with Aiding and Abetting EoCs for constantly putting these tiles down. Its a combat nightmare (Speed = Winning in SS13) and make chasing antagonists/targets/etc on our larger maps a nightmare.

Changing it to a QoL item instead of a ZOOOOOOOOOM item retains its usefulness for Xenobio to make without making every player have to play Sonic the Hedgehog Simulator for any sort of chase.

## Testing
Ran about like a chicken with my head cut off.

## Changelog
:cl:
tweak: Bluespace Tiles now affect slowdown instead of increasing your movespeed.
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
There are only two pieces of gear I can see that would get de-speed'd due to this:

- Ninja Shoes
- The Admin Suit

Considering these are both admin-only, I frankly do not care for them.

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 1]
```diff
@@ -5,7 +5,11 @@
 	. += dna.species.movement_delay(src)
 	if(isobj(pulling) && has_gravity(pulling))
 		var/obj/pulled = pulling
-		. += pulled.pull_speed
```
[the remark this answers is not in the document]
Comment (matttheficus) [reply code/modules/mob/living/carbon/human/human_movement.dm:8]: Outta scope for this PR, would need to be in a PR dedicated to pulling stuff.

File: code/modules/mob/living/carbon/human/species/_species.dm [hunk 2]
```diff
@@ -289,21 +289,21 @@
...
-		ADD_SLOWDOWN(H.gloves.slowdown)
-	if(!H.buckled && H.shoes)
-		ADD_SLOWDOWN(H.shoes.slowdown)
-	if(H.back)
-		ADD_SLOWDOWN(H.back.slowdown)
-	if(H.l_hand && (H.l_hand.flags & HANDSLOW))
-		ADD_SLOWDOWN(H.l_hand.slowdown)
-	if(H.r_hand && (H.r_hand.flags & HANDSLOW))
-		ADD_SLOWDOWN(H.r_hand.slowdown)
+	if(!istype(get_turf(H), /turf/simulated/floor/bluespace))
```
[the remark this answers is not in the document]
Comment (matttheficus) [reply code/modules/mob/living/carbon/human/species/_species.dm:292]: Thought a bit more on traits. Unsure if I want to add a trait that EVERY SINGLE MOB has to check/have for this to work. Doing a type check is fine here IMO (also cleaned the pulling one up a bit). If for some reason there's more of a push toward using traits instead from Headcoders/PRRs Ill look into swapping it.

[the remark this answers is not in the document]
Comment (GDNgit) [reply code/modules/mob/living/carbon/human/species/_species.dm:292]: Matt you can put the trait on the turf

[the remark this answers is not in the document]
Comment (matttheficus) [reply code/modules/mob/living/carbon/human/species/_species.dm:292]: `if(!has_trait(get_turf(H), TRAIT_WHATEVER))` is basically the exact same kind of check IMO - just replacing an `istype` with a `has_trait`. I dont see a massive benefit from one to the other.

File: code/game/turfs/simulated/floor.dm [hunk 3]
```diff
@@ -34,6 +34,7 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 	var/barefootstep = FOOTSTEP_HARD_BAREFOOT
 	var/clawfootstep = FOOTSTEP_HARD_CLAW
 	var/heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	var/buffs
```
Comment (Dimach) [inline code/game/turfs/simulated/floor.dm:37]:
```suggestion
```
Useless code

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 4]
```diff
@@ -4,6 +4,7 @@
 	. += GLOB.configuration.movement.human_delay
 	. += dna.species.movement_delay(src)
 
+
 /mob/living/carbon/human/Process_Spacemove(movement_dir = 0)
```
Comment (Henri215) [inline code/modules/mob/living/carbon/human/human_movement.dm:8]:
```suggestion
	. += dna.species.movement_delay(src)

/mob/living/carbon/human/Process_Spacemove(movement_dir = 0)
```

File: code/__HELPERS/trait_helpers.dm [hunk 5]
```diff
@@ -298,3 +299,10 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_CANNOT_PULL "pullblocked"
 /// Abstract condition that prevents movement if being pulled and might be resisted against. Handcuffs and straight jackets, basically.
 #define TRAIT_RESTRAINED "restrained"
+
+//***** TURF TRAITS *****//
+/// Removes slowdown while walking on these tiles.
+#define TRAIT_BLUESPACE_SPEED "bluespace_speed_trait"
```
Comment (Contrabang) [inline code/__HELPERS/trait_helpers.dm:305]: Add this to the global init list of `traits_by_type` please

File: code/modules/mob/living/carbon/human/species/_species.dm [hunk 6]
```diff
@@ -290,20 +290,22 @@
...
-	if(!H.buckled && H.shoes)
-		ADD_SLOWDOWN(H.shoes.slowdown)
-	if(H.back)
-		ADD_SLOWDOWN(H.back.slowdown)
-	if(H.l_hand && (H.l_hand.flags & HANDSLOW))
-		ADD_SLOWDOWN(H.l_hand.slowdown)
-	if(H.r_hand && (H.r_hand.flags & HANDSLOW))
-		ADD_SLOWDOWN(H.r_hand.slowdown)
+	var/turf/simulated/floor/T = get_turf(H)
+	if(!HAS_TRAIT(T, TRAIT_BLUESPACE_SPEED))
```
Comment (S34NW) [inline code/modules/mob/living/carbon/human/species/_species.dm:294]:
```suggestion
	if(istype(T) && !HAS_TRAIT(T, TRAIT_BLUESPACE_SPEED))
```
No point typecasting if we don't check the result

**Review by S34NW — CHANGES_REQUESTED** [review]
One thing then good for merge

**Review by S34NW — APPROVED** [review]

**Pull request merged**